src/Subtitles/SubtitleComponent.js

```diff
@@ -0,0 +1,203 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Lightning from '../Lightning'
+
+export default class SubtitleComponent extends Lightning.Component {
+  static _template() {
+    return {
+      visible: false,
+      rect: true,
+      color: 0x90000000,
+      shader: { type: Lightning.shaders.RoundedRectangle, radius: 5 },
+      Text: {
+        y: 5,
+        x: 20,
+        text: {
+          textColor: 0xffffffff,
+          fontSize: 38,
+          lineHeight: 38 * 1.4,
+          textAlign: 'center',
+          wordWrap: true,
+          maxLines: 3,
+          shadow: true,
+          shadowColor: 0xff333333,
+        },
+      },
+    }
+  }
+
+  _init() {
+    this._textTextureDefaults = new Lightning.textures.TextTexture(this.stage).cloneArgs()
+
+    this.tag('Text').on('txLoaded', ({ _source }) => {
+      this.w = _source.w + this.tag('Text').x * 2
+      this.h = _source.h
+      this.position()
+    })
+  }
+
+  get textFormat() {
+    const textTag = this.tag('Text').text
+    return {
+      fontFace: textTag.fontFace || 'sans-serif',
+      fontSize: textTag.fontSize,
+      lineHeight: textTag.lineHeight,
+      textAlign: textTag.textAlign,
+      wordWrap: true,
+      maxLines: textTag.maxLines,
+    }
+  }
```

Comment:
Should the textColor be also returned here?

Reply:
There is no need for the text color to be returned here. It's not relevant in the pre-render calculation of the text texture. We're only interested in the size of the texture and the color doesn't affect that